mobile/library/kotlin/io/envoyproxy/envoymobile/grpc/GRPCStream.kt

```diff
@@ -51,9 +51,19 @@ class GRPCStream(
   }
 
   /**
-   * Close this connection.
+   * Close the stream, wait for the peer to finish before actually closing the stream.
    */
   fun close() {
+    // The gRPC protocol requires the client stream to close with a DATA frame.
+    // More information here:
+    // https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests
     underlyingStream.close(ByteBuffer.allocate(0))
   }
+
+  /**
+   * Cancel the stream forcefully regardless of whether the peer has more data to send.
+   */
+  fun cancel() {
```

Comment:
mildly prefer reset stream but given there's already an API for cancel I'm going to say it's fine as-is :-P

Reply:
speaking from the perspective of mobile dev I decided to use `cancel` verb in here as `URLSession` and `OkHttp` networking libraries have a concept of cancelling a request while they do not have a concept of resetting a request. That does not mean that we _need_ to use `cancel` in here but in general I think that this name will be more user-friendly for its target group.